dirsrvtests/tests/suites/replication/regression_test.py

```diff
@@ -1054,6 +1056,294 @@ def test_online_init_should_create_keepalive_entries(topo_m2):
+        9. No error and Initialization done.
+        10. No error
+        11. c1 time skew should be higher than threshold
+
+    """
+
+    # Variables initialization
+    repl = ReplicationManager(DEFAULT_SUFFIX)
+
+    m1 = topo_m2c2.ms["master1"]
+    m2 = topo_m2c2.ms["master2"]
+    c1 = topo_m2c2.cs["consumer1"]
+    c2 = topo_m2c2.cs["consumer2"]
+
+    replica_m1 = Replicas(m1).get(DEFAULT_SUFFIX)
+    replica_m2 = Replicas(m2).get(DEFAULT_SUFFIX)
+    replica_c1 = Replicas(c1).get(DEFAULT_SUFFIX)
+    replica_c2 = Replicas(c2).get(DEFAULT_SUFFIX)
+
+    replicid_m2 = replica_m2.get_rid()
+
+    agmts_m1 = Agreements(m1, replica_m1.dn)
+    agmts_m2 = Agreements(m2, replica_m2.dn)
+
+    m1_m2 = get_agreement(agmts_m1, m2)
+    m1_c1 = get_agreement(agmts_m1, c1)
+    m1_c2 = get_agreement(agmts_m1, c2)
+    m2_m1 = get_agreement(agmts_m2, m1)
+    m2_c1 = get_agreement(agmts_m2, c1)
+    m2_c2 = get_agreement(agmts_m2, c2)
+
+    # Step 1: Disable m1<->m2 agreement to avoid propagate timeSkew
+    m1_m2.pause()
+    m2_m1.pause()
+
+    # Step 2: Generate ldif without replication data
+    m1.stop()
+    m2.stop()
+    ldif_file = '%s/norepl.ldif' % m1.get_ldif_dir()
+    m1.db2ldif(bename=DEFAULT_BENAME, suffixes=[DEFAULT_SUFFIX],
+               excludeSuffixes=None, repl_data=False,
+               outputfile=ldif_file, encrypt=False)
+    # Remove replication metadata that are still in the ldif
+    # _remove_replication_data(ldif_file)
+
+    # Step 3: Increase time skew on master2
+    timeSkew=6*3600
+    # We can modify master2 time skew
+    # But the time skew on the consumer may be smaller
+    # depending on when the cnsgen generation time is updated
+    # and when first csn get replicated.
+    # Since we use timeSkew has threshold value to detect
+    # whether there are time skew or not,
+    # lets add a significative margin (longer than the test duration)
+    # to avoid any risk of erroneous failure
+    timeSkewMargin = 300
+    DSEldif(m2)._increaseTimeSkew(DEFAULT_SUFFIX, timeSkew+timeSkewMargin)
+
+    # Step 4: Init both masters from that ldif
+    m1.ldif2db(DEFAULT_BENAME, None, None, None, ldif_file)
+    m2.ldif2db(DEFAULT_BENAME, None, None, None, ldif_file)
+    m1.start()
+    m2.start()
+
+    # Step 5: Perform on line init from master1 to consumer1
+    #          and from master2 to consumer2
+    m1_c1.begin_reinit()
+    m2_c2.begin_reinit()
+    (done, error) = m1_c1.wait_reinit()
+    assert done is True
+    assert error is False
+    (done, error) = m2_c2.wait_reinit()
+    assert done is True
+    assert error is False
+
+    # Step 6: Perform update on both masters
+    repl.test_replication(m1, c1)
+    repl.test_replication(m2, c2)
+
+    # Step 7: Check that c1 has no time skew
+    # Stop server to insure that dse.ldif is uptodate
+    c1.stop()
+    c1_nsState = DSEldif(c1).readNsState(DEFAULT_SUFFIX)[0]
+    c1_timeSkew = int(c1_nsState['time_skew'])
```

Comment:
With time_skew, is it possible that it could occur given some other system clock change during the test?

Reply:
Ahhh you set the time_skew manually. I still would be curious if it could be affected by other system affects though.

Reply:
As the test is based on time skew, yes changing  the system time while the test is running would affect the result.
   ( moving the time backwards may increase the final time skew and moving it forward may decrease it) 
That said with a timeSkewMargin of 5 minutes on a test that spends less than 2 minutes even a poorly written NTP software should not break the test.
And I suspect that quite a number of tests would be broken by a significant system time change.

Reply:
We have pytest-libfaketime integration that can be used for time change shenanigans (see `dirsrvtests/tests/suites/healthcheck/health_security_test.py` for example).
And for the server part I had an unfinished integration for libfaketime with lib389 somewhere. It was editing the systemd drop-in config file to load libfaketime using LD_PRELOAD and control it using resource file. Something like here: https://bugzilla.redhat.com/show_bug.cgi?id=1843157#c4 
It has issues with SELinux though, so it needs some work to become a part of our tests.
I need to find it and submit.

Reply:
Thank you for the information. that is interesting to know. 
Using libfaketime to change the result of time syscall within master2 is indeed another way to increase the csn generator time skew. 
That said it has exactly the same behavior than tweaking the csngen state (i.e test result may be wrong if global system time change by more than 5 minutes).
And due to the facts that lib389 already deals with csngen state internal to compute the time skew and that there are still issue around libfaketime solution, I propose to kept the csngen state tweak solution.

Reply:
Yes, it's just for information. I agree with your approach, it looks good to me.